### ISSUE TITLE
Constant permissions errors as players are using the module

### ISSUE THREAD

**Issue by mr-bo-jangles** [herasrobert/icu5e#6]
I'm assuming this is because you don't currently have a way for users to hide/show tokens with their current set of permissions, even as "Trusted" users.

What you could do to work around this limitation for now is: 
1. Add a number of Macros (Show token, Show Hidden Door, etc) as part of the module, probably via a Compendium.
2. Have something to import them automatically on install, or have a button that imports them in settings as "Execute as GM" macros.
3. In your code, use these macros whenever the players trigger an action.
4. Inside the macros, do your own permissions checks, like checking for the presence of a GM level user on the scene being modified, etc. If these fail, raise a message saying that permission was denied because the GM wasn't online/present on the scene/whatever.

**Comment by herasrobert** [comment]
Interesting bug and I really like your idea for a solution. Do you know if
the permissions error occur all the time during normal use or is an
experimental features enabled? It'll help me narrow it down a bit in the
code.

**Comment by mr-bo-jangles** [comment]
It's certainly prevelant with the experimental features turned on, but it's triggered by anytime a player users executes the following line https://github.com/herasrobert/icu5e/blob/a9841aab471c4d8713c6f6913dd83082c77ba7b2/scripts/icu5e.js#L376

**Comment by mr-bo-jangles** [comment]
I should also say, that it's probably present in the changing of visibility of secret walls and doors, but I've not used those features yet. I have also noticed that the issue can be additionally worked around from the GM's perspective if they have the token making perception checks selected, as it appears that both the GM and the token owner run the code surrounding un-hiding the hostile tokens

**Comment by JWJRogers** [comment]
ive been struggling with this my self.
(love this module by the way)
i was wondering if creating a custome "ownership level" could get this working right?
just giving players the behind scene power to hide/unhide tokens (or what ever the powers are) but nothing else so they cant just see the monsters.
this version works if i give the player full ownership and works once on lesser levels except "none"
oh this happens with and without exsperimental features in use.
ty for this module by the way
Error Log Message:-
{"level":"error","timestamp":"2020-11-30 09:07:48","message":"User test lacks permission to update Tokens in parent t1DkpnZ43R4HijnA","stack":"Error: User test lacks permission to update Tokens in parent t1DkpnZ43R4HijnA\n    at Function._onUpdate (D:\\Program Files (x86)\\FoundryVTT\\resources\\app\\dist\\database\\odm\\embedded.js:1:7963)\n    at async Socket.handleEvent (D:\\Program Files (x86)\\FoundryVTT\\resources\\app\\dist\\sockets.js:1:2294)"}

**Comment by herasrobert** [comment]
Hi guys, I'm sorry but I haven't had a chance to work on this the past few months and likely won't in the future. I had spoken to someone with a similar module to see if they wanted to take it over but that didn't pan out.

**Comment by JWJRogers** [comment]
Hay don't worry
Life is life at the moment 
Just survive 
I'm not coder at all but know some one who is. 
If there up for it do you mind if I try and get a work around done?

**Comment by herasrobert** [comment]
Not at all, please do whatever you want to do. I've always loved the idea
of open source and people contributing to projects.

Originally the project started for my need to have my players passive
perception reveal enemies, traps in the form of tokens, and secret doors.
